unified-scheduler-pool/src/lib.rs

```diff
@@ -569,8 +569,7 @@ mod chained_channel {
 
     // P doesn't need to be `: Clone`, yet rustc derive can't handle it.
     // see https://github.com/rust-lang/rust/issues/26925
-    #[derive(Derivative)]
-    #[derivative(Clone(bound = "C: Clone"))]
+    #[derive_where(Clone)]
```

Comment:
thoughts on just implementing `Clone` here instead of adding another dependency that may become stale in the future. It seems really straight-forward:

```rust
impl<P, C: Clone> Clone for ChainedChannelReceiver<P, C> {
    fn clone(&self) -> Self {
        Self {
            reciever: self.receiver.clone(),
            aux_receiver: self.aux_receiver.clone(),
            context: self.context.clone(),
        }
    }
} 
```

Reply:
@apfitzge sorry for the delay. took some time to consolidate my thoughts with some research.

in short, I'd like to avoid manual impl and use this new crate (called `derive-where` among others) for following _personal preferences_:
- i'm generally inclined to using crates as much as possible because I'm open-source maxi and upstreaming-pr happy guy :)
- i'm moderately against boilerplate code, which is error-prone, imo. yes, to recite, I'm saying this, even if we're talking about straight-forward manual impl of just 9 LoC. (less code == less maintenance; in fact, upcoming pr would have to tweak the manual impl.)
- oftentimes, crate shopping (shh, my hobby) could refresh one's rust knowledge. so, i hope this habit could be rewarded in some way. for example, i noticed `ImmutableDeserializedPacket` can be simplified.
- in this particular case, switching cost is very small, should the new one could be stale (unlike say, `bincode`, which would break our wire-format should we switch).
- i generally think (slightly) longer build time by adding another dep can generally be offset by increased productivity or concise code base in this case.
- thanks to rust opt, all tested clone impls (std, manual, those crates mentioned on this pr) generates the same machine code.

Reply:
😄 `ImmutableDeserializedPacket` is awful and is going to go away entirely, which is the simplest it can become.